synapse/lib/node.py

```diff
@@ -327,8 +327,19 @@ async def set(self, name, valu, init=False):
             if curv == norm:
                 return False
 
-        props = {prop.name: norm}
-        nodeedits = await self.snap.getNodeAdds(self.form, self.ndef[1], props, addnode=False)
+        # Optimize fast property setting when the destination property is
+        # not a form and when normalizing the property we have no subs
+        # or adds to consider.
+        if self.snap.core.model.form(prop.type.name) is None \
```

Comment:
Lets functionalize this logic so we can avoid the ugly concatenations and deep nesting. Something like:

```
def _isFastPathProp(self, prop, info):
    if self.snap.core.model.form(prop.type.name) is not None:
        return False
    if info.get('subs') is not None:
        return False
    ... etc etc ...
    return(True)
```